### DISCUSSION TITLE
Handle cancel on input type file

### DISCUSSION THREAD

**Discussion by VictorBoca** [react-hook-form/react-hook-form#3730] [Q&A] [answered]
Hi

I have a form which contain a file, if I add a file and after that I upload again, I doesn't select any file and press cancel, the file is reseted, the old file is gone. It is there a way to avoid this issue, after a small investigation the onChange is triggered also if the event.target.files.length is  0

Here it is an example

    <form onSubmit={handleSubmit(onSubmit)}>
       <input name={'firstName'} ref={register({ required: true })} />
       <input name={'lastName'} ref={register({ required: true })} />
       <input name={'profilePicture'} type={'file'} ref={register({ required: true })} />
    </form>
    
    
"react-hook-form": 6.9.2

**Comment by bluebill1049** [comment]
not quite sure what's your question above, but the input file is always going to be sensitive input due to security.

**Reply by VictorBoca** [reply]
Hi

I took a small video demo, might help to understand exactly the issue that I'm facing 😃 

If I already have a file chosen and I want to change it, but I hit **cancel** instead of  **open** the old open file is gone.

https://user-images.githubusercontent.com/57251570/102683433-a82a8680-41d9-11eb-988d-90ff18636a2a.mov

**Comment by VictorBoca** [comment, marked as answer]
Hello, my bad, after I read more carefully the docs I understand that the files have to be handled using the Controlled component and fire the onChange event only if files are selected

`onChange={event => {
              if (event?.target?.files.length > 0) 
                props.onChange(event.target.files);
            }}`
